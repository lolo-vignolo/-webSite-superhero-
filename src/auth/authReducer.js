import React from 'react'
import { types } from '../types/types';

//estado si el usuario esta autenticado
// const state = {
//     name: "Lorenzo",
//     logged: true
// }


// const ejAction = {

//     type : "login",
//     payload : {
//         name : "newname",
//         logged : true or false
//     }
// }


export const authReducer = (state = {}, action ) => {
   

    switch (action.type) {
        case  types.login:
            
            return {
              ...action.payload,
              logged : true
          }
          
        case types.logout:
            
            return{
                logged:false
            }

        default:
        return state;



    }
}

