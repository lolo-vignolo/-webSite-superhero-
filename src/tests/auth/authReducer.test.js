
import { authReducer } from "../../auth/authReducer"
import { types } from "../../types/types"

describe('Puebas en el authReducer ', () => {

    //es una funcion, no es un componente por eso no debo usar shallow

    test('should shows the default statement ', () => {

        const estadoInicial = {
            logged:false
        }

        const result = authReducer({estadoInicial}, {})

        expect (result).toEqual({estadoInicial})

        
    })


    test('should autenticate the name and user statement', () => {

        // por defaul el type.login me devuelve un objeto con name y type, agregandole el payload name, cambio ese name.
        const accion = {
            type: types.login,
            payload : {
                name : "Alfon"
            }
            
            }
        
               const state = authReducer({logged:false}, accion)


               expect (state).toEqual({logged:true , name : "Alfon"})
    })

    test('should autenticate the name and user statement', () => {

        // por defaul el type.login me devuelve un objeto con name y type, agregandole el payload name, cambio ese name.
        const accion = {
            type: types.logout
            }
        
               const state = authReducer({logged:true , name : "Alfon"}, accion)


               expect (state).toEqual({logged:false })
    })




    
    
    
})
