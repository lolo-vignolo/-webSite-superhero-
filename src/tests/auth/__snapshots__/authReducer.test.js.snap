// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en el authReducer  should shows the default statement  1`] = `
<authReducer
  state={
    Object {
      "logged": true,
      "name": "Lorenzo",
    }
  }
/>
`;
